File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, ArrayList<String>> map = new HashMap<>();
        for (int i = 0; i < strs.length; i++) {
            char[] chararray = strs[i].toCharArray();
            Arrays.sort(chararray);
            String word = new String(chararray);
            if (!map.containsKey(word)) {
                map.put(word, new ArrayList<String>());
            } 
            map.get(word).add(strs[i]);
        }
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;

        for (auto i{0uz}; i < strs.size(); ++i) {
            auto tmp = strs[i];
            sort(tmp.begin(), tmp.end());
            groups[tmp].push_back(strs[i]);
        }

        vector<vector<string>> res;

        for (auto& [_, value] : groups) {
            res.push_back(value);
        }

        return res;
    }
};